lib/rubocop/cop/rails/three_state_boolean.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces that boolean columns are created with default values and
+      # NOT NULL constraint.
+      #
+      # @example
+      #   # bad
+      #   add_column :users, :active, :boolean
+      #   t.column :active, :boolean
+      #   t.boolean :active
+      #
+      #   # good
+      #   add_column :users, :active, :boolean, default: true, null: false
+      #   t.column :active, :boolean, default: true, null: false
+      #   t.boolean :active, default: true, null: false
+      #
+      # @example StartAfterMigrationVersion: 20220301000001
+      #   # bad
+      #   # db/migrate/20220301000002_add_active_to_users.rb
+      #   add_column :users, :active, :boolean
+      #
+      #   # good
+      #   # db/migrate/20220301000001_add_active_to_users.rb
+      #   add_column :users, :active, :boolean
+      #
+      class ThreeStateBoolean < Base
+        include StartAfterMigrationVersion
+
+        MSG = 'Boolean columns should always have a default value and a NOT NULL constraint.'
+
+        RESTRICT_ON_SEND = %i[add_column column boolean].freeze
+
+        def_node_matcher :three_state_boolean?, <<~PATTERN
+          {
+            (send nil? :add_column _ _ {(sym :boolean) (str "boolean")} $_ ?)
+            (send !nil? :column _ {(sym :boolean) (str "boolean")} $_ ?)
+            (send !nil? :boolean _ $_ ?)
+          }
+        PATTERN
+
+        def_node_matcher :required_options?, <<~PATTERN
+          (hash <(pair (sym :default) !nil?) (pair (sym :null) false) ...>)
+        PATTERN
+
+        def on_send(node)
+          return unless starts_after_migration_version?
+
+          three_state_boolean?(node) do |options_node|
+            options_node = options_node.first
+
+            add_offense(node) unless required_options?(options_node)
```

Comment:
It would be awesome if this was compatible with the `change_column_default` approach suggested at https://github.com/ankane/strong_migrations#adding-a-column-with-a-default-value

Reply:
One idea: if the migration file also contains the string `change_column_default` or `change_column_null` assume they are following strong migrations advice and don't warn.

Reply:
Implemented this suggestion. 
Having only `change_column_null` should be enough since this will already avoid nulls and 3-state booleans, despite having default or not.

Reply:
I think, it is very possible for someone to do everything in the same migration: https://github.com/fatkodima/online_migrations#example

> So I think we do actually need to check for change_column_default.

> assume they are following strong migrations advice and don't warn.

I would like not to assume the user will add not null in the following migration.